src/access/universal.test.ts

```diff
@@ -0,0 +1,889 @@
+    );
+    const mockedResource = mockSolidDatasetFrom(
+      "https://arbitrary.pod/resource"
+    );
+    const mockedResourceWithAcl = addMockResourceAclTo(mockedResource);
+    getResourceInfoWithAcr.mockResolvedValueOnce(mockedResourceWithAcl as any);
+    const wacGetPublicAccess = jest.spyOn(wacModule, "getPublicAccess");
+    wacGetPublicAccess.mockResolvedValue(null);
+    const options = { fetch: () => new Promise(jest.fn()) as any };
+
+    await getPublicAccess("https://arbitrary.pod/resource", options);
+
+    expect(wacGetPublicAccess).toHaveBeenCalledWith(expect.anything(), options);
+  });
+
+  it("returns null if the Resource has neither an ACR nor an ACL", async () => {
+    const getResourceInfoWithAcr = jest.spyOn(
+      acpLowLevel,
+      "getResourceInfoWithAcr"
+    );
+    const mockedResource = mockSolidDatasetFrom(
+      "https://arbitrary.pod/resource"
+    );
+    getResourceInfoWithAcr.mockResolvedValueOnce(mockedResource as any);
+
+    const access = await getPublicAccess("https://arbitrary.pod/resource");
+
+    expect(access).toBeNull();
+  });
+});
+
+describe("setPublicAccess", () => {
+  it("calls out to the well-tested ACP API for Resources with an ACR", async () => {
+    const getResourceInfoWithAcr = jest.spyOn(
+      acpLowLevel,
+      "getResourceInfoWithAcr"
+    );
+    const mockedResource = mockSolidDatasetFrom(
+      "https://arbitrary.pod/resource"
+    );
+    const mockedAcr = mockAcrFor("https://arbitrary.pod/resource");
+    const mockedResourceWithAcr = addMockAcrTo(mockedResource, mockedAcr);
+    getResourceInfoWithAcr.mockResolvedValueOnce(mockedResourceWithAcr);
+    jest.spyOn(acpModule, "internal_setPublicAccess");
+
+    await setPublicAccess("https://arbitrary.pod/resource", { read: true });
+
+    expect(acpModule.internal_setPublicAccess).toHaveBeenCalledTimes(1);
+  });
+
+  it("returns the new effective access for ACP-controlled Resources", async () => {
+    const getResourceInfoWithAcr = jest.spyOn(
+      acpLowLevel,
+      "getResourceInfoWithAcr"
+    );
+    const mockedResource = mockSolidDatasetFrom(
+      "https://arbitrary.pod/resource"
+    );
+    const mockedAcr = mockAcrFor("https://arbitrary.pod/resource");
+    const mockedResourceWithAcr = addMockAcrTo(mockedResource, mockedAcr);
+    getResourceInfoWithAcr.mockResolvedValueOnce(mockedResourceWithAcr);
+    jest
+      .spyOn(acpModule, "internal_setPublicAccess")
+      .mockReturnValueOnce(mockedResourceWithAcr);
+    jest
+      .spyOn(acpLowLevel, "saveAcrFor")
+      .mockResolvedValueOnce(undefined as any);
+    jest
+      .spyOn(acpModule, "internal_getPublicAccess")
+      .mockReturnValueOnce({ append: true });
+
+    const returnedAccess = await setPublicAccess(
+      "https://arbitrary.pod/resource",
+      { read: true }
+    );
+
+    expect(returnedAccess).toStrictEqual({ append: true });
+  });
+
+  it("calls out to the well-tested WAC API for Resources with an ACR", async () => {
```

Comment:
```suggestion
  it("calls out to the well-tested WAC API for Resources with an ACL", async () => {
```

Reply:
30f7c937

(that you actually went through the effort of pointing them all out - I'd just have said "etc." :P)